torch/csrc/distributed/rpc/rref.h

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+#include <c10/util/Optional.h>
+#include <torch/csrc/distributed/rpc/message.h>
+#include <torch/csrc/distributed/rpc/rpc_agent.h>
+#include <torch/csrc/distributed/rpc/types.h>
+
+#include <atomic>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+class RRef;
+class RRefContext;
+class UserRRef;
+
+// Represents fork of an RRef to be sent over the wire.
+//
+// In order to preserve correctness of reference counting, each RRefForkData
```

Comment:
Yes, they are enforced at a higher layer, but anyone using `RRefForkData` should be aware. They should not create multiple RRefs from `RRefForkData` (this can be done through `RRefContext`), and they should not throw away `RRefForkData` they got from `fork()` (because the user could have notified the owner on the fork).